### PR TITLE
Fix React typings

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Just like @gaetanmaisse suggested (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38961#issuecomment-539646061), I updated all React imports from `import React from 'react'` to `import * as  React from 'react'`, but I also tried to import as much as possible so the `* as React` part isn't necessary.

I left the stories and tests just like they were.

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@gaetanmaisse Can you point me in the right direction for fixing the errors please?
It seems like they're not related to this PR.

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
I'll take a look at why this fails

**Comment by MichaelDeBoey** [comment]
Thanks @ndelangen!

I see errors about missing `@emotion/native` types, but these were already missing before this PR

**Comment by ndelangen** [comment]
The problem does not exist on the next branch.

It is something that has changed in this PR that is causing this.

```
declare module '@emotion/native';
```

Is still in `typings.d.ts` so I'm confused why it's not picking this up.

**Comment by MichaelDeBoey** [comment]
Hmmm very strange indeed then.

The only thing I've done in this PR changing all `React.*` calls and just import the things directly.

`React.createRef` -> `createRef`

File: addons/a11y/src/components/ColorBlindness.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { document } from 'global';
-import React, { ReactNode, useState } from 'react';
+import React, { FunctionComponent, ReactNode, useState } from 'react';
```
Comment (gaetanmaisse) [inline addons/a11y/src/components/ColorBlindness.tsx]: With `import React, { FunctionComponent, ReactNode, useState } from 'react';` we are still importing React using default syntax I think you should use `import * as React from 'react';` and `React.FunctionComponent` etc everytime `import React from 'react';` is still needed.

Comment (gaetanmaisse) [reply addons/a11y/src/components/ColorBlindness.tsx]: hmmm, in fact, is importing React still needed?

Comment (MichaelDeBoey) [reply addons/a11y/src/components/ColorBlindness.tsx]: It's indeed needed for the TSX files.

**Comment by ndelangen** [comment]
I think I fixed it

**Comment by MichaelDeBoey** [comment]
Hi @ndelangen!

Thanks for fixing types!

Looking at your fixes, these weren't PR related were they? 🤔

**Comment by ndelangen** [comment]
Well something caused typescript to stop loading in the `typings.d.ts`.

I found a way to include it back, and found some improvements to make typing stricter in the RN app.

**Pull request merged**